services/ymax-planner/src/main.ts

```diff
@@ -275,11 +283,18 @@ export const main = async (
     gasEstimator,
     usdcTokensByChain,
   };
-  await startEngine(powers, {
-    isDryRun,
-    contractInstance: config.contractInstance,
-    depositBrandName: env.DEPOSIT_BRAND_NAME || 'USDC',
-    feeBrandName: env.FEE_BRAND_NAME || 'BLD',
+
+  await withDeferredCleanup(async addCleanup => {
+    await startEngine(powers, {
+      isDryRun,
+      contractInstance: config.contractInstance,
+      depositBrandName: env.DEPOSIT_BRAND_NAME || 'USDC',
+      feeBrandName: env.FEE_BRAND_NAME || 'BLD',
+    });
+    addCleanup(async () => {
+      console.log('fraz close');
+      await db.close();
+    });
```

Comment:
It does no good to register a cleanup after `await`ing the wrapped function.
```suggestion
  await withDeferredCleanup(async addCleanup => {
    addCleanup(() => db.close());
    await startEngine(powers, {
      isDryRun,
      contractInstance: config.contractInstance,
      depositBrandName: env.DEPOSIT_BRAND_NAME || 'USDC',
      feeBrandName: env.FEE_BRAND_NAME || 'BLD',
    });
```